tests/test_defects.py

```diff
@@ -0,0 +1,270 @@
+        self.assertEqual(defects, merged)
+
+        # now start manipulating defect lists
+        config = copy.copy(self.defaultConfig)
+        config.combinationMode = 'FRACTION'
+        config.combinationFraction = 0.85
+        task = cpPipe.defects.FindDefectsTask(config=config)
+        merged = task._postProcessDefectSets([defects, defects], self.flatExp.getDimensions(), 'FRACTION')
+
+        defectList = [defects]*10  # 10 identical defect sets
+        # remove one defect from one of them, should still be over threshold
+        defectList[7] = defectList[7][:-1]
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'FRACTION')
+        self.assertEqual(defects, merged)
+
+        # remove another and should be under threshold
+        defectList[3] = defectList[3][:-1]
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'FRACTION')
+        self.assertNotEqual(defects, merged)
+
+        # now test the AND and OR modes
+        defectList = [defects]*10  # 10 identical defect sets
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'AND')
+        self.assertEqual(defects, merged)
+
+        defectList[7] = defectList[7][:-1]
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'AND')
+        self.assertNotEqual(defects, merged)
+
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'OR')
+        self.assertEqual(defects, merged)
+
+    def test_pixelCounting(self):
+        config = copy.copy(self.defaultConfig)
+        config.nPixBorderUpDown = 0
+        config.nPixBorderLeftRight = 0
+        task = cpPipe.defects.FindDefectsTask(config=config)
+        defects = task.findHotAndColdPixels(self.flatExp, 'flat')
+
+        defectArea = 0
+        for defect in defects:
+            defectArea += defect.getBBox().getArea()
+
+        crossCheck = 0
+        for x, y, s in self.brightDefects:
+            crossCheck += s**2
+        for x, y, s in self.darkDefects:
+            crossCheck += s**2
+
+        self.assertEqual(defectArea, task._nPixFromDefects(defects))  # the real function being tested
+        self.assertEqual(defectArea, crossCheck)  # but calculate the sum two ways
+
+    def test_getNumGoodPixels(self):
+        testImage = self.flatExp.clone()
+        mi = testImage.maskedImage
+
+        imageSize = testImage.getBBox().getArea()
+        nGood = self.defaultTask._getNumGoodPixels(mi)
+
+        self.assertEqual(imageSize, nGood)
+
+        NODATABIT = mi.mask.getPlaneBitMask("NO_DATA")
+
+        noDataBox = Box2I(Point2I(31, 49), Extent2I(3, 6))
+        testImage.mask[noDataBox] |= NODATABIT
+
+        self.assertEqual(imageSize - noDataBox.getArea(), self.defaultTask._getNumGoodPixels(mi))
+        self.assertEqual(imageSize, self.defaultTask._getNumGoodPixels(mi, 'BAD'))  # check for misfire
```

Comment:
"misfire" of what?

Reply:
Have clarified in the comment, which now reads
`# check for misfire; we're setting NO_DATA here, not BAD`